.circleci/real_config.yml

```diff
@@ -112,85 +112,74 @@ commands:
+    steps:
+      - skip-if-dir:
+          dir: docs
 
   skip-if-not-docs:
     steps:
-      - run: git remote add upstream https://github.com/determined-ai/determined
-      - run: tools/scripts/retry.sh git fetch upstream
-      - run:
-          name: check for any docs changes
-          command: |
-            MERGE_BASE="$(git merge-base upstream/main HEAD)"
-            DIFF_DIRS="$(git diff-tree --no-commit-id --name-only "$MERGE_BASE" HEAD)"
-            # check "shared" is in the list of changed directories
-            if [[ "$DIFF_DIRS" == *"docs"* ]]; then
-              echo "docs change detected, running downstream steps"
-            else
-              echo "detected non-docs changes, halting job now"
-              circleci-agent step halt
-            fi
+      - skip-if-dir:
+          dir: docs
+          not: true
 
   skip-if-webui-only:
     steps:
-      - run: tools/scripts/retry.sh git fetch upstream
-      - run:
-          name: check for webui-only changes # must be run after skip-if-docs-only
-          command: |
-            MERGE_BASE="$(git merge-base upstream/main HEAD)"
-            DIFF_DIRS="$(git diff-tree --no-commit-id --name-only "$MERGE_BASE" HEAD)"
-            if [ "$DIFF_DIRS" = "webui" ] ; then
-              echo "webui-only change detected, halting job now"
-              circleci-agent step halt
-            else
-              echo "detected non-webui-only changes: $DIFF_DIRS"
-            fi
+      - skip-if-dir:
+          dir: webui
 
   skip-if-not-shared:
     steps:
-      - run: git remote add upstream https://github.com/determined-ai/determined
-      - run: tools/scripts/retry.sh git fetch upstream
-      - run:
-          name: check for shared changes
-          command: |
-            MERGE_BASE="$(git merge-base upstream/main HEAD)"
-            DIFF_DIRS="$(git diff-tree --no-commit-id --name-only "$MERGE_BASE":webui/react/src HEAD:webui/react/src)"
-            # check "shared" is in the list of changed directories
-            if [[ "$DIFF_DIRS" == *"shared"* ]]; then
-              echo "shared code changed, running downstream test"
-            else
-              echo "shared code not affected, halting job now"
-              circleci-agent step halt
-            fi
+      - skip-if-dir:
+          dir: shared
+          not: true
 
   skip-if-github-only:
     steps:
-      - run: tools/scripts/retry.sh git fetch upstream
-      - run:
-          name: check for github-only changes
-          command: |
-            MERGE_BASE="$(git merge-base upstream/main HEAD)"
-            DIFF_DIRS="$(git diff-tree --no-commit-id --name-only "$MERGE_BASE" HEAD)"
-            if [ "$DIFF_DIRS" = ".github" ] ; then
-              echo "github-only change detected, halting job now"
-              circleci-agent step halt
-            else
-              echo "detected changes outside of .github: $DIFF_DIRS"
-            fi
+      - skip-if-dir:
+          dir: .github
```

Comment:
I don't think this is an equivalent change.  There will always be a .github directory in a workdir with a checkout of main - that's part of the repo.  The logic which was removed does a diff of the current version object against the main branch, and then checks to see if the output string contains only .github.  That would indicate that the current branch only changed something like codeonwers or a workflow, so CI can be skipped.

The replacement logic I think will just skip every CI run.  That'll save money, but result in less effective testing. :)

Reply:
...and I should've reviewed from the top down instead of the bottom up.  I saw "skip-if" and interpreted that as a "when" - not something new. :sweat: